### PR TITLE
parser: Add shorthand for `this.`

### PR DESCRIPTION
**Pull request by awesomekling**
After writing a couple hundred lines of Jakt, the main thing that
stood out was the large amount of `this.` spam throughout methods.

Let's try to mitigate that with a shorthand for refering to member
variables: `^foo` is short for `this.foo`

### CODE REVIEW

**Comment by mustafaquraish** [comment]
for whatever my 2 cents are worth, `^` seems like a bit of a weird choice for `this`, makes it feel like it's some weird unary operator when looking at it. Just bikeshedding now, but I feel like perhaps `$` might be nicer? So something like `$foo` or `$.foo`. Maybe even `@`?

Or... maybe I'll just get used to `^`, but this is just my first impression

**Comment by linusg** [comment]
`^` is usually XOR, I'm against this.

**Comment by awesomekling** [comment]
Sure, let's roll with `$`, that was my initial choice as well. :)

Please note that this is an experiment to see if we can improve the code feel. If we end up disliking it, we can always change it.

**Comment by mustafaquraish** [comment]
> Sure, let's roll with `$`, that was my initial choice as well. :)
> 
> Please note that this is an experiment to see if we can improve the code feel. If we end up disliking it, we can always change it.

Looks much nicer to me, but yeah lets see what it feels like!

PS: the test file is still called `this_caret.jakt`

File: src/parser.rs [hunk 1]
```diff
@@ -2423,6 +2423,27 @@ pub fn parse_operand(tokens: &[Token], index: &mut usize) -> (ParsedExpression,
...
+                    *index += 1;
+                    ParsedExpression::IndexedStruct(
+                        Box::new(this_expr),
+                        name.clone(),
+                        tokens[*index].span,
+                    )
+                }
+                _ => {
+                    error = error.or(Some(JaktError::ParserError(
+                        "Missing member name after '^'".to_string(),
```
Comment (linusg) [inline src/parser.rs:2440]:
```suggestion
                        "Missing member name after '$'".to_string(),
```

**Review by linusg — CHANGES_REQUESTED** [review]

File: src/parser.rs (hunk 1 above)
Comment (linusg) [reply src/parser.rs:2440]: Probably should have a test as well ;^)

File: README.md [hunk 2]
```diff
@@ -178,6 +178,10 @@ let mutable foo = Foo(x: 3)
 foo.set(9)
 ```
 
+### Shorthand for accessing member variables
+
+To reduce repetitive `this.` spam in methods, the shorthand `^foo` expands to `this.foo`.
```
Comment (mustafaquraish) [inline README.md:183]: `$foo`

**Pull request merged**